eth_abi/abi.py

```diff
@@ -1,28 +1,44 @@
-"""
-Vendored from `pyethereum.abi`
-"""
-from eth_utils import (
-    is_bytes,
+from typing import (
+    Any,
+    Iterable,
+    Tuple,
 )
 
+from eth_utils import is_bytes
+
 from eth_abi.decoding import (
     ContextFramesBytesIO,
     TupleDecoder,
 )
 
-from eth_abi.exceptions import EncodingError
-
 from eth_abi.encoding import TupleEncoder
 
+from eth_abi.exceptions import EncodingError
+
 from eth_abi.registry import registry
 
+from eth_abi.typing import (
```

Comment:
Can you move these to a separate python module called `eth_typing`.  Taking a look in the py-evm repo for how it's done there.  We're planning on extracting these to a common repo that can be shared across our codebases.

I think this should make migrating to this new repo once it's been created easier.

Reply:
Sure, no problem.

Reply:
So that would be a separate module which is also installed by setup.py when installing eth-abi?  It doesn't seem like the installer would gracefully manage merging different versions of that module into the import hierarchy.  For example, wouldn't my version of `eth_typing/__init__.py` would differ from py-evm's version?